### ISSUE TITLE
Datacite DOIs partly broken

### ISSUE THREAD

**Issue by adam3smith** [zotero/translators#2018]
I have not investigated how widely that's the case, but for at least some Datacite DOIs, the `schemaVersion` has been removed and thus [our detect](https://github.com/zotero/translators/blob/master/Datacite%20JSON.js#L55) (and hence import) fails. I've [reported this upstream](https://github.com/datacite/lupo/issues/347)  at Datacite. I'd suggest we wait for a couple of days before changing the translator as the schema is by far the most elegant way to detect.

**Comment by adam3smith** [comment]
DataCite have confirmed verbally that they'll fix this.

**Comment by Jmuccigr** [comment]
How can I tell which DOIs are affected? I've got a couple that resolve just fine via the doi.org URI, but not with Zotero.

**Comment by alexis-michaud** [comment]
Hi, just a note to thank for the information on what's going wrong here and confirm that the fix will be hailed with much enthusiasm. The [Pangloss Collection](https://pangloss.cnrs.fr/), an open archive of endangered languages, now has DOIs for all resources, with serious metadata (code available [here](https://github.com/vasaura/DoiPangloss)), and seamless integration with Zotero would be super-cool. 

(Here's one such DOI, if anyone wants to test:  https://doi.org/10.24397/pangloss-0005709 )

**Comment by alexis-michaud** [comment]
> How can I tell which DOIs are affected? I've got a couple that resolve just fine via the doi.org URI, but not with Zotero.

@Jmuccigr : You can't tell from looking at the DOI: you need to know if the database where the corresponding metadata are storied is at DataCite or CrossRef. If the former, Zotero import won't work until the bug mentioned above is fixed at their end.

**Comment by alexis-michaud** [comment]
Hi, just a note to thank for the information on what's going wrong here and confirm that the fix will be hailed with much enthusiasm. The [Pangloss Collection](https://pangloss.cnrs.fr/), an open archive of endangered languages, now has DOIs for all resources, with serious metadata (code available [here](https://github.com/vasaura/DoiPangloss)), and seamless integration with Zotero would be super-cool. 

(Here's one such DOI, if anyone wants to test:  https://doi.org/10.24397/pangloss-0005709 )

**Comment by adam3smith** [comment]
> How can I tell which DOIs are affected? 

If you use https://search.datacite.org/works/{doi} in your browser and it shows a search result, then it's a datacite DOI. So for the ones above:
https://search.datacite.org/works/10.24397/pangloss-0005709

**Comment by alexis-michaud** [comment]
👍

**Comment by bjohas** [comment]
Would it be at all possible to follow up DataCite? We'd really like to use Zenodo DOIs for import to Zotero... https://forums.zotero.org/discussion/comment/353170

**Comment by alexis-michaud** [comment]
DataCite Technical Director Martin Fenner kindly answered an e-mail query sent to him on this topic saying:
"this is an open bug that we will hopefully fix soon enough" (April 6th, 2020).
We too look forward very much to the fix.

**Comment by mfenner** [comment]
@alexis-michaud thanks for posting this. Because we are dealing with a number of other important issues at DataCite right now, I am reluctant to give a specific timeline for a fix. But we will aim to resolve this in April.

**Comment by alexis-michaud** [comment]
👍

**Comment by dstillman** [comment]
@adam3smith: This seems to be causing a freeze in PDF metadata retrieval. We should obviously fix that in Zotero itself, but as a quick hack that we can push out right away, do you think we can change https://github.com/zotero/translators/blob/ec2eaef36131c631c997ab05c6ec9491d482c0a6/DOI%20Content%20Negotiation.js#L87 to just look for `"agency": "DataCite"`? (Trying to parse the text as JSON and access the property would be cleaner but probably not necessary. Obviously checking the response Content-Type would be best but we can't do that currently.)

**Comment by mfenner** [comment]
Unfortunately "agency" is not yet available for all DOIs in our API, so would probably also not reliably work. We have some other fields that are probably unique (e.g. ClientId, ProviderId, state), but schema and agency are really the two fields that make the most sense. 

Until the end of the year we will be adding many DOIs from other registration agencies to our index, and we have already added 8.5 million Crossref DOIs. As you can see in https://api.datacite.org/dois?client-id=crossref.citations, all these Crossref DOIs use schema `http://datacite.org/schema/kernel-4`, as we convert the Crossref metadata into DataCite XML. So thinking about it a bit more, `agency` is the best field to determine whether a DOI from our APIs is a DataCite DOI.

Obviously you fetch metadata via DOI content negotiation so should never run into the Crossref DOIs in our index. One question I have is why you don't directly ask for Citeproc JSON from DataCite in the DOI content negotiation, the implementation is much better than a few years ago.

To summarize, what is the best way to fix this? Should DataCite fix the inclusion of the schema "http://datacite.org/schema" as quickly as possible, but definitely before the end of April, or do you want to switch to another strategy to detect a DataCite DOI, e.g. via `agency`?

**Comment by adam3smith** [comment]
> One question I have is why you don't directly ask for Citeproc JSON from DataCite in the DOI content negotiation, the implementation is much better than a few years ago.

I don't remember the details, but we experimented with this in January 2019 and found that we got significantly better import using the Datacite JSON

**Comment by dstillman** [comment]
OK, I put in a temporary fix for Retrieve Metadata for PDF. The DOI Content Negotation translator is now looking for `"agency": "DataCite"` or `"providerId":` in addition to `http://datacite.org/schema`.

Unfortunately that wasn't sufficient, because at least for 10.7916/D8959HR1, the DataCite JSON import then failed with `data.subjects is undefined` looking for tags, and if I put that in a conditional it failed with `data.formats is undefined` looking for `medium`. Others would have failed after that, so I just made most of the fields optional.

For reference, here's the request we're making:

`curl -v -L -H "Accept: application/vnd.datacite.datacite+json, application/vnd.crossref.unixref+xml, application/vnd.citationstyles.csl+json" 'https://doi.org/10.7916%2Fd8959hr1'`

When I tried just removing `application/vnd.datacite.datacite+json` from the `Accept` header, DataCite returned a 200 with `Content-Type: application/vnd.crossref.unixref+xml` and no content. It only worked if I removed Crossref XML from `Accept` as well, in which case it returned CSL JSON.

But for now, with the above fixes, the DataCite JSON is working, at least for this DOI.

**Comment by adam3smith** [comment]
Best I can tell this is also fixed at DataCite - closing
